l10n_it_delivery_note/controllers/portal.py

```diff
@@ -0,0 +1,132 @@
+from odoo import _
+from odoo.exceptions import AccessError, MissingError
+from odoo.http import request, route
+
+from odoo.addons.portal.controllers.portal import CustomerPortal, pager as portal_pager
+
+
+class DNCustomerPortal(CustomerPortal):
+    def _get_delivery_note_domain(self, search_in=False):
+        domain = [("state", "in", ["confirm", "invoiced", "done"])]
+        if search_in:
+            domain += search_in
+        return domain
```

Comment:
Segnalo che, senza ulteriori filtri, gli utenti "interni" (Sia Mitchell Admin che Marc Demo) vedono tutte le Delivery Note nel portale che potrebbero vedere dal backend, che appartengano loro o meno. Mi sembra di vedere che anche altri oggetti funzionano allo stesso modo, per quanto mi sembri un comportamento strano magari va bene così.

Reply:
Non so se ho capito bene cosa intendi, però di default mi sembra che le DN non hanno permessi in base agli utenti che le hanno create o dei partner impostati.
Attualmente quindi il portale rispecchierebbe il comportamento del backend e non penso sia da modificare in questa PR, al massimo si può aprire una issue per discuterne.
@francesco-ooops ?

Reply:
@odooNextev lo allinerei al comportamento dei SO nel portale, quali sono visibili per gli utenti interni?

Reply:
> @odooNextev lo allinerei al comportamento dei SO nel portale, quali sono visibili per gli utenti interni?

attualmente nel portale vengono mostrate quelle visibili da backend

Reply:
Allora concordo che è al di fuori dello scopo della modifica

Reply:
Per gli utenti portale, il funzionamento è lo stesso del SO, e mi sembra corretto

Per gli utenti interni credo non ci sia una vera necessità di accedere ai DN dal portale, io rimuoverei la voce dal portale per loro se è fattibile

Reply:
> Per gli utenti interni credo non ci sia una vera necessità di accedere ai DN dal portale, io rimuoverei la voce dal portale per loro se è fattibile

@aleuffre @GSLabIt condividete l'idea di @francesco-ooops di rimuovere la voce delle DN per gli utenti interni dal portale?

Reply:
A meno che non ci siano ragioni forti per fare diversamente, ero più d'accordo con la prima idea di allineare il più possibile al comportamento dei SO su Odoo base, per quanto risulti anche a me un comportamento strano.

Reply:
> 

Il problema è che i DN non sono regolati con un diritto di accesso relativo ai propri documenti o quelli altrui, per cui il risultato è che un utente con diritto di accesso "SO: documenti personali" e senza "Magazzino: utente" potrebbe vedere tutti i DN relativi a tutti i SO, e se i DN contengono i prezzi praticamente diventa un modo per vedere tutti i SO

A livello funzionale, non trovo fondamentale che un utente con accesso al backend usi il frontend per accedere ai documenti

Reply:
Purtroppo non è così (test su runboat), perchè nei seguenti casi non c'è nulla che blocca la visualizzazione in frontend: 

> un utente senza accesso al magazzino non può accedere in backend ai picking e di conseguenza al DN

> un utente "vendite: personali" e "magazzino: utente" può accedere solo ai DN relazionati ai propri ordini (non penso possa accedere ai DN relazionati ad ordini non suoi)

Reply:
Chiaro. Sono comunque d'accordo a non replicare la funzionalità qui, e nel caso in altra PR migliorare le regole di accesso dei DN.